homeassistant/helpers/entity_platform.py

```diff
@@ -653,6 +645,16 @@ async def _async_add_entity(  # noqa: C901
             entity.add_to_platform_abort()
             return
 
+        if entity.registry_entry and entity.registry_entry.disabled:
+            self.logger.debug(
+                "Not adding entity %s because it's disabled",
+                entry.name
+                or entity.name
+                or f'"{self.platform_name} {entity.unique_id}"',
+            )
+            entity.add_to_platform_abort()
+            return
+
```

Comment:
Add a test to test_entity_platform.py which asserts the new behavior:
- unique_id collision has priority over registry disabled
- the registry entry is cleared if there's a unique_id collision